controllers/openstackclient_controller.go

```diff
@@ -94,17 +94,63 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	// check for required secrets
 	_, _, err = common.GetSecret(r, instance.Spec.DeploymentSSHSecret, instance.Namespace)
-	if err != nil && errors.IsNotFound(err) {
+	if err != nil && k8s_errors.IsNotFound(err) {
 		return ctrl.Result{RequeueAfter: time.Second * 20}, fmt.Errorf("DeploymentSSHSecret secret does not exist: %v", err)
 	} else if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	if instance.Status.OpenStackClientNetStatus == nil {
+		instance.Status.OpenStackClientNetStatus = map[string]ospdirectorv1beta1.HostStatus{}
+	}
+
+	//
+	// create hostname for the openstackclient
+	//
+	hostnameRef := instance.GetHostnames()
+	hostnameDetails := common.Hostname{}
+	if len(hostnameRef) == 0 {
+		hostnameDetails = common.Hostname{
+			Basename: openstackclient.Role,
+			Hostname: "",
+			VIP:      false,
+		}
+
+		err = common.CreateOrGetHostname(instance, &hostnameDetails)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		for hostname := range hostnameRef {
+
+			hostnameDetails = common.Hostname{
```

Comment:
It looks like `hostnameDetails` will always end up being the last element of `hostnameRef`.  Just want to make sure that's what you intended

Reply:
for openstackclient we always only have 0 or 1 hostnames.